pmd-core/src/main/java/net/sourceforge/pmd/renderers/AbstractRenderer.java

```diff
@@ -62,6 +71,47 @@ public void setShowSuppressedViolations(boolean showSuppressedViolations) {
         this.showSuppressedViolations = showSuppressedViolations;
     }
 
+    @Override
+    public void setUseShortNames(List<String> inputPaths) {
+        this.inputPathPrefixes = inputPaths;
+    }
+
+    /**
+     * Determines the filename that should be used in the report depending on the
+     * option "shortnames". If the option is enabled, then the filename in the report
+     * is without the directory prefix of the directories, that have been analyzed.
+     * If the option "shortnames" is not enabled, then the inputFileName is returned as-is.
+     *
+     * @param inputFileName
+     * @return
+     *
+     * @see PMDConfiguration#isReportShortNames()
+     * @see PMDParameters#isShortnames()
+     */
+    protected String determineFileName(String inputFileName) {
+        for (final String prefix : inputPathPrefixes) {
+            final Path prefPath = Paths.get(prefix).toAbsolutePath();
+            final String prefPathString = prefPath.toString();
+
+            if (inputFileName.startsWith(prefPathString)) {
+                if (prefPath.toFile().isDirectory()) {
+                    return trimAnyPathSep(inputFileName.substring(prefPathString.length()));
+                } else {
+                    if (inputFileName.indexOf(FILE_SEPARATOR.charAt(0)) == -1) {
```

Comment:
or `File.separatorChar` directly
```suggestion
                    if (inputFileName.indexOf(File.separatorChar) == -1) {
```